### PR TITLE
change syntax

### PR DESCRIPTION
**Pull request by DawidWesierski4**
This pull requests shows the "fragmented" approach where prase Mail have 3 helping functions this approach may be more visible

### CODE REVIEW

File: main.c [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+#include "parseMail.h"
+
+void
+automaticStatusTestParseMail(void)
+{
+   struct automaticStatusTestStruct {
+      const char* inputName;
+      int expectedResult;
+   };
+   int i;
```
Comment (pgardocx) [inline main.c]: empty line

File: main.c [hunk 2]
```diff
@@ -0,0 +1,139 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include "parseMail.h"
+
+void
+automaticStatusTestParseMail(void)
+{
+   struct automaticStatusTestStruct {
+      const char* inputName;
```
Comment (pgardocx) [inline main.c]: const char *inputName;

File: main.c [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+      printf("FAIL\n");
+   }
+}
+
+void
+unitTest(const char *one)
+{
+   credentials janK;
+   int aux;
+   aux = ParseMail(one, &janK);
```
Comment (pgardocx) [inline main.c]: can be in one line

File: main.c [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+   if (!aux) {
```
Comment (pgardocx) [inline main.c]: empty line

File: main.c [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+      printCredentials(janK);
+   } else {
+      printf("Test: %s Failed \n",one);
+      printf("RESULT: %d\n\n", aux);
+   }
+   releaseCredentialsMemory(&janK);
```
Comment (pgardocx) [inline main.c]: empty line

File: parseMail.c [hunk 6]
```diff
@@ -45,149 +72,164 @@ checkName(const char *name)
...
-      fprintf(stderr, "ERR_INPUT_FORMAT\n");
+      printErrorMsg(ERR_INPUT_FORMAT);
       return ERR_INPUT_FORMAT;
    }
 
+   outPtr = ((credentials*)out);
    atPosition = atCharPtr - mail;
    dotCharPtr = strchr(mail, '.');
-
    if (dotCharPtr && atPosition > dotCharPtr - mail) {
```
Comment (pgardocx) [inline parseMail.c:204]: empty line

File: main.c (hunk 2 above)
Comment (DawidWesierski4) [reply main.c]: DONE

File: main.c (hunk 1 above)
Comment (DawidWesierski4) [reply main.c]: DONE

File: main.c (hunk 3 above)
Comment (DawidWesierski4) [reply main.c]: DONE

File: main.c (hunk 4 above)
Comment (DawidWesierski4) [reply main.c]: DONE

File: main.c (hunk 5 above)
Comment (DawidWesierski4) [reply main.c]: DONE

File: parseMail.c (hunk 6 above)
Comment (DawidWesierski4) [reply parseMail.c:204]: DONE

File: parseMail.c [hunk 7]
```diff
@@ -1,193 +1,271 @@
...
+}
+
+/* parses mail surname and domain for the ParseMail function */
+int
+parseName(const char **mail, const char *charPtr, char **outPtr,
+          int (*checkFunction)(const char*))
+{
+   int check, size = charPtr - *mail;
+
+   if(!mail || !charPtr || !outPtr) {
```
Comment (pgardocx) [inline parseMail.c]: space after if

File: parseMail.h [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+#include <stdbool.h>
+
+typedef struct credentialInformation {
+   char *name;
+   char *surName;
+   char *domain;
+} credentials;
+
+enum ERR_NUMS {
+   ERR_SUCCESSFUL,
```
Comment (pgardocx) [inline parseMail.h]: maybe let's call it just SUCCESS, "ERR" is a little bit misleading

File: parseMail.h [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+   ERR_ALLOCATING_MEMORY,
+   ERR_NULL_POINTER,
+   ERR_INPUT_FORMAT,
+   ERR_INPUT_FORMAT_SIZE_RULE_VIOLATION,
+   ERR_INPUT_FORMAT_ILLEGAL_CHAR,
+};
+
+extern bool parseMailWarningEnabled;
```
Comment (pgardocx) [inline parseMail.h]: please hide it only in .c file (static) and add functions for enabling/disabling logs

File: parseMail.c (hunk 7 above)
Comment (DawidWesierski4) [reply parseMail.c]: DONE

File: parseMail.h (hunk 8 above)
Comment (DawidWesierski4) [reply parseMail.h]: DONE

File: parseMail.h (hunk 9 above)
Comment (DawidWesierski4) [reply parseMail.h]: DONE

**Pull request merged**